dali/base/dadfs.cpp

```diff
@@ -9567,7 +9571,7 @@ class CDaliDFSServer: public Thread, public CTransactionLogTracker, implements I
         StringArray authScopes;
         CIArrayOf<CFileMatch> matchingFiles;
         start = msTick();
-        count = scanner.getResults(auth, udesc, matchingFiles, authScopes);
+        count = scanner.getResults(auth, udesc, matchingFiles, authScopes, false);
```

Comment:
I'd imagine most of the memory is used before the sorting routes are called, i.e. by the serialized files themselves.
Perhaps esp doesn't currently make calls that end up calling iterateFiles?